p2panda-rs/src/test_utils/mocks/node.rs

```diff
@@ -495,6 +495,18 @@ impl Node {
         document.view().to_owned()
     }
 
+    pub fn get_document_but_dont_make_it_look_like_a_view(&self, id: &Hash) -> Document {
```

Comment:
You mentioned this in the chat I think, that `DocumentView` doesn't contain everything needed when construction `KeyGroup`s. Is that right? I'd say there might be two reasons/solutions for this:

1) the mock node implementation needs changing, and should be returning `Document`s in most cases
2) actually we need to rethink what goes where in `Document` and `DocumentView`

When `Document` was being born, I made a decision at some point to contain all meta data there, and keep `DocumentView` very simple. The idea being, is that `Document` is what we pass around, and we access the document view from it. Happy to rethink this if it doesn't fit well with `KeyGroup` construction patterns, which are definitely more complex than what we've had so far.

Reply:
I think 1) would be quite nice as `Document` has everything we need here already.

Reply:
https://github.com/p2panda/p2panda/issues/286